chips/esp32-c3/src/sysreg.rs

```diff
@@ -0,0 +1,106 @@
+//! TimG Group driver.
+
+use kernel::utilities::registers::interfaces::{ReadWriteable, Readable};
+use kernel::utilities::registers::{register_bitfields, register_structs, ReadWrite};
+use kernel::utilities::StaticRef;
+
+pub const SYS_REG_BASE: StaticRef<SysRegRegisters> =
+    unsafe { StaticRef::new(0x600C_0000 as *const SysRegRegisters) };
+
+register_structs! {
+    pub SysRegRegisters {
+        (0x000 => _reserved0),
+        (0x008 => cpu_per_conf: ReadWrite<u32, CPU_PER_CONF::Register>),
+        (0x00c => _reserved1),
+        (0x010 => perip_clk_en0: ReadWrite<u32, PERIP_CLK_EN0::Register>),
+        (0x014 => _reserved3),
+        (0x058 => sysclk_config: ReadWrite<u32, SYSCLK_CONFIG::Register>),
+        (0x05C => _reserved_unimplemented_yet),
+        (0x1000 => @END),
+    }
+}
+
+register_bitfields![u32,
+    PERIP_CLK_EN0 [
+        TIMERGROUP0 OFFSET(13) NUMBITS(1) []
+    ],
+    CPU_PER_CONF [
+        CPUPERIOD_SEL OFFSET(0) NUMBITS(2) [
+            MHz80 = 0,
+            MHz160 = 1,
+        ],
+        PLL_FREQ_SEL OFFSET(2) NUMBITS(1) [
+            MHz320 = 0,
+            MHz480 = 1
+        ],
+        CPU_WAIT_MODE_FORCE_ON OFFSET(3) NUMBITS(1) [],
+        CPU_WAIT_DELAY_NUM OFFSET(4) NUMBITS(4) [],
+    ],
+    SYSCLK_CONFIG [
+        PRE_DIV_CNT OFFSET(0) NUMBITS(10) [],
+        SOC_CLK_SEL OFFSET(10) NUMBITS(2) [
+            Xtal = 0,
+            Pll = 1,
+            Fosc = 2
+        ],
+        CLK_XTAL_FREQ OFFSET(12) NUMBITS(6) [],
+    ]
+];
+
+#[repr(u8)]
```

Comment:
why `#[repr(u8)]`? It looks like this is just cast to a u32 in the one place it is used anyway, and forcing a u8 representation can actually lead to extra code for casting to/from other widths, and can prevent certain niche optimizations.

Reply:
You are right, I modified it to u32.